src/main/app-updater.ts

```diff
@@ -1,19 +1,170 @@
+import logger from "./logger";
+import dateFormat from "dateformat";
+import { broadcastIpc, IpcChannel, NotificationChannelAdd, NotificationChannelPrefix } from "../common/ipc";
+import { ipcMain } from "electron";
+import { isDevelopment } from "../common/vars";
+import { SemVer } from "semver";
 
-export default class AppUpdater {
+function delay(duration: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, duration));
+}
+
+class NotificationBackchannel {
+  private static _id = 0;
+
+  static nextId(): IpcChannel {
+    return `${NotificationChannelPrefix}${NotificationBackchannel._id++}`
+  }
+}
+
+const title = "Lens Updater";
+
+async function autoUpdateNow(): Promise<void> {
+  const body = "Downloading and installing update.";
+  broadcastIpc({
+    channel: NotificationChannelAdd,
+    args: [{
+      title,
+      body,
+      status: "info",
+      timeout: 5000,
+    }]
+  })
+
+  logger.info("[UPDATE CHECKER]: update downloaded started");
+  await autoUpdater.downloadUpdate();
+  logger.info("[UPDATE CHECKER]: update downloadeded");
+  autoUpdater.quitAndInstall();
+}
+
+async function autoUpdateCheck(args: UpdateInfo): Promise<void> {
+  return new Promise(async resolve => {
+    const body = "Install and restart Lens?";
+    const yesNowChannel = NotificationBackchannel.nextId();
+    const yesLaterChannel = NotificationBackchannel.nextId();
+    const noChannel = NotificationBackchannel.nextId();
 
-  protected updateInterval: number = (1000 * 60 * 60 * 24) // once a day
+    function cleanupChannels() {
+      ipcMain.removeAllListeners(yesNowChannel);
+      ipcMain.removeAllListeners(yesLaterChannel);
+      ipcMain.removeAllListeners(noChannel);
+    }
 
-  constructor() {
-    autoUpdater.logger = logger
+    ipcMain
+      .on(yesNowChannel, async () => {
+        cleanupChannels();
+
+        await autoUpdater.downloadUpdate();
+        autoUpdater.quitAndInstall();
+
+        resolve();
+      })
+      .on(yesLaterChannel, async () => {
+        cleanupChannels();
+
+        await autoUpdater.downloadUpdate();
+        autoUpdater.autoInstallOnAppQuit = true;
+
+        resolve();
+      })
+      .on(noChannel, () => {
+        cleanupChannels();
+        resolve();
+      });
+
+    broadcastIpc({
+      channel: NotificationChannelAdd,
+      args: [{
+        title,
+        body,
+        status: "info",
+        buttons: [
+          {
+            label: "Yes, now",
+            backchannel: yesNowChannel,
+            style: {
+              background: "green",
+              marginRight: "10px"
+            }
+          },
+          {
+            label: "Yes, on quit",
+            backchannel: yesLaterChannel,
+            style: {
+              background: "green",
+              marginRight: "10px"
+            }
+          },
+          {
+            label: "No",
+            backchannel: noChannel,
+            accent: true
+          }
+        ],
```

Comment:
Styles and translatable texts better to keep in renderer components.

Reply:
Styles yes, I can move those. However, I don't see why we should have the text in the renderer since its source is `main`.

Reply:
It doesn't matter where the "source" of texts. It's a part of layout now (especially when you show notifications in HTML-block).
For `i18n` support it's easier to keep it in `.tsx` for extracting to dictionary.
But, nevermind, localization support will required quite a lot of changes anyway.